cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -26,15 +27,64 @@ import (
 	apiv1 "k8s.io/kubernetes/pkg/api/v1"
 )
 
-// AwsCloudProvider implements CloudProvider interface.
-type AwsCloudProvider struct {
+// awsCloudProvider implements CloudProvider interface.
+type awsCloudProvider struct {
 	awsManager *AwsManager
 	asgs       []*Asg
 }
 
 // BuildAwsCloudProvider builds CloudProvider implementation for AWS.
-func BuildAwsCloudProvider(awsManager *AwsManager, specs []string) (*AwsCloudProvider, error) {
-	aws := &AwsCloudProvider{
+func BuildAwsCloudProvider(awsManager *AwsManager, discoveryOpts cloudprovider.NodeGroupDiscoveryOptions) (cloudprovider.CloudProvider, error) {
+	staticDiscoverySpecified := len(discoveryOpts.NodeGroupSpecs) > 0
+	autoDiscoverySpecified := discoveryOpts.NodeGroupAutoDiscoverySpec != ""
+	if staticDiscoverySpecified && autoDiscoverySpecified {
+		return nil, errors.New("Failed to build an aws cloud provider: Either node group specs or node group auto discovery spec can be specified but not both")
```

Comment:
Can we include the flag values in this error message (`--node-group-auto-discovery` or `--nodes`)